File: app/src/main/java/com/example/sunnyjain/todo3/di/ActivityBuilder.java
package com.example.sunnyjain.todo3.di;

import com.example.sunnyjain.todo3.MainActivity;
import com.example.sunnyjain.todo3.MainActivityModule;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBuilder {
    @ContributesAndroidInjector(modules = MainActivityModule.class)
    abstract MainActivity bindMainActivity();

}

File: app/src/main/java/com/example/sunnyjain/todo3/di/ActivityBuilder.kt
package com.example.sunnyjain.todo3.di;

import com.example.sunnyjain.todo3.MainActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector
@Suppress("unused")
@Module
abstract class ActivityBuilder {
    @ContributesAndroidInjector(modules = [FragmentBuilderModule::class])
    abstract fun contributeMainActivity(): MainActivity

}
